openmc/tallies.py

```diff
@@ -1482,19 +1482,30 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Create copies of self and other tallies to rearrange for tally
+        # arithmetic
+        self_copy = copy.deepcopy(self)
+        other_copy = copy.deepcopy(other)
+
         # Find any shared filters between the two tallies
-        self_filters = set(self.filters)
-        other_filters = set(other.filters)
+        self_filters = set(self_copy.filters)
+        other_filters = set(other_copy.filters)
         filter_intersect = self_filters.intersection(other_filters)
 
-        # Align the shared filters to follow in each tally operand
+        # Align the shared filters in successive order
```

Comment:
When you start looking into why this works for Python 2 but not 3, you might consider checking `filter_intersect`. I'm curious if the `Filters` in it are sorted in the same order in Python 2 and 3 - if not, that might be the issue behind the test suite failures.